rewriter/RequiresAncestorKernel.cc

```diff
@@ -0,0 +1,59 @@
+#include "rewriter/RequiresAncestorKernel.h"
+#include "ast/Helpers.h"
+#include "ast/ast.h"
+#include "core/Names.h"
+#include "core/core.h"
+
+using namespace std;
+
+namespace sorbet::rewriter {
+
+namespace {
+
+bool hasAnyRequiresAncestorCall(core::MutableContext ctx, ast::ClassDef *klass) {
+    for (auto &stat : klass->rhs) {
+        if (auto send = ast::cast_tree<ast::Send>(stat)) {
+            if (send->fun == core::Names::requiresAncestor() && send->recv.isSelfReference() && send->hasBlock()) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+} // namespace
+
+void RequiresAncestorKernel::run(core::MutableContext ctx, ast::ClassDef *klass) {
+    // Only run for Module definitions
+    if (klass->kind != ast::ClassDef::Kind::Module) {
+        return;
+    }
+
+    // Check if requires_ancestor feature is enabled
+    if (!ctx.state.cacheSensitiveOptions.requiresAncestorEnabled) {
+        return;
+    }
+
+    // If any requires_ancestor call is present, do nothing (regardless of its body)
+    if (hasAnyRequiresAncestorCall(ctx, klass)) {
+        return;
+    }
+
+    // Add extend T::Helpers and requires_ancestor { Kernel } declarations
+    auto loc = klass->loc;
+    auto locZero = loc.copyWithZeroLength();
+
+    // Add extend T::Helpers
+    klass->rhs.emplace_back(ast::MK::Send1(loc, ast::MK::Self(loc), core::Names::extend(), locZero,
+                                           ast::MK::Constant(loc, core::Symbols::T_Helpers())));
+
+    // Add requires_ancestor { Kernel }
+    auto kernelConstant = ast::MK::UnresolvedConstant(loc, ast::MK::EmptyTree(), core::Names::Constants::Kernel());
```

Comment:
This can be a resolved constant, right? We always mean `::Kernel`, not any other possible (though unlikely) nested `Kernel`.

```suggestion
    // Add requires_ancestor { ::Kernel }
    auto kernelConstant = ast::MK::Constant(loc, ast::MK::EmptyTree(), core::Names::Constants::Kernel());
```

Reply:
I think rewriters run before the resolve phase, so I don't think anything would be resolved at this stage.

Reply:
Ah, ok, that makes sense. Will update to use that instead. Thanks